### ISSUE TITLE
access_entities of non standard type incompatible with module

### ISSUE THREAD

**Issue by Bourne-ID** [terraform-aws-modules/terraform-aws-eks#2893]
## Description

This module expects the variable access_entries to have a nested key of "policy_associations" even if the access entity being created is incompatible with policies, like EC2_Linux or other similar types. The flattening logic ignores these entries, therefore it is not possible to create these access_entries types through this module

This variable should be compatible with types like EC2_LINUX. 
 

- [X] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: 2.0.0

- Terraform version:1.5.5

- Provider version(s): 5.35.0

## Reproduction Code [Required]
(using example/eks_managed_node_group, extended with)

```HCL
data "aws_partition" "current" {}

locals {
  partition = data.aws_partition.current.partition

  access_entries = {
    role_standard_expected = {
      principal_arn       = aws_iam_role.roles[0].arn
      type                = "STANDARD"
      policy_associations = {
        admin = {
          policy_arn   = "arn:${local.partition}:eks::aws:cluster-access-policy/AmazonEKSClusterAdminPolicy"
          access_scope = {
            type       = "namespace"
            namespaces = ["bourneid"]
          }
        }
      }
    },
    role_ec2_missing = {
      principal_arn       = aws_iam_role.roles[1].arn
      type                = "EC2_LINUX"
      // policy_association is not possible with this type
    }
  }
}

resource "aws_iam_role" "roles" {
  count = 2
  assume_role_policy = jsonencode({
    "Version": "2012-10-17",
    "Statement": [
      {
        "Sid": "",
        "Effect": "Allow",
        "Principal": {
          "Service": "ec2.amazonaws.com"
        },
        "Action": "sts:AssumeRole"
      }
    ]
  })
}


```

Plan result:
```HCL
Terraform will perform the following actions:

  # module.eks.aws_eks_access_entry.this["role_standard_expected_admin"] will be created
  + resource "aws_eks_access_entry" "this" {
      + access_entry_arn  = (known after apply)
      + cluster_name      = "ex-eks-managed-node-group"
      + created_at        = (known after apply)
      + id                = (known after apply)
      + kubernetes_groups = (known after apply)
      + modified_at       = (known after apply)
      + principal_arn     = "arn:aws:iam::[accountid]:role/terraform-20240203055027812300000002"
      + tags              = {
          + "Example"    = "ex-eks-managed-node-group"
          + "GithubOrg"  = "terraform-aws-modules"
          + "GithubRepo" = "terraform-aws-eks"
        }
      + tags_all          = {
          + "Example"    = "ex-eks-managed-node-group"
          + "GithubOrg"  = "terraform-aws-modules"
          + "GithubRepo" = "terraform-aws-eks"
        }
      + type              = "STANDARD"
      + user_name         = (known after apply)
    }

  # module.eks.aws_eks_access_policy_association.this["role_standard_expected_admin"] will be created
  + resource "aws_eks_access_policy_association" "this" {
      + associated_at = (known after apply)
      + cluster_name  = "ex-eks-managed-node-group"
      + id            = (known after apply)
      + modified_at   = (known after apply)
      + policy_arn    = "arn:aws:eks::aws:cluster-access-policy/AmazonEKSClusterAdminPolicy"
      + principal_arn = "arn:aws:iam::[accountid]:role/terraform-20240203055027812300000002"

      + access_scope {
          + namespaces = [
              + "bourneid",
            ]
          + type       = "namespace"
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  ~ access_entries                                   = {
      + role_standard_expected_admin = {
          + access_entry_arn  = (known after apply)
          + cluster_name      = "ex-eks-managed-node-group"
          + created_at        = (known after apply)
          + id                = (known after apply)
          + kubernetes_groups = (known after apply)
          + modified_at       = (known after apply)
          + principal_arn     = "arn:aws:iam::[accountid]:role/terraform-20240203055027812300000002"
          + tags              = {
              + Example    = "ex-eks-managed-node-group"
              + GithubOrg  = "terraform-aws-modules"
              + GithubRepo = "terraform-aws-eks"
            }
          + tags_all          = {
              + Example    = "ex-eks-managed-node-group"
              + GithubOrg  = "terraform-aws-modules"
              + GithubRepo = "terraform-aws-eks"
            }
          + timeouts          = null
          + type              = "STANDARD"
          + user_name         = (known after apply)
        }
    }

```

* Generate roles required
* Create map to pass to module
* set `access_entries = local.access_entries`
* plan/apply

## Expected behavior

2 access entities should have been created - one for each role. 1 access policy association is expected. 

## Actual behavior

only 1 access entity is being created for the STANDARD role, however the EC2_LINUX is being excluded. 

## Additional context

The issue appears to be around 
```HCL
flattened_access_entries = flatten([
    for entry_key, entry_val in local.merged_access_entries : [
      for pol_key, pol_val in lookup(entry_val, "policy_associations", {}) :
```
Where if the policy_associations is not set, a blank map is used by default. This prevents the nested for loop from firing, and therefore no content is generated for access_entries without this key. 

Attempts to bypass this by setting up a blank policy_association fails with
```HCL
│ Error: Unsupported attribute
│ 
│   on ../../main.tf line 213, in resource "aws_eks_access_policy_association" "this":
│  213:   policy_arn    = each.value.association_policy_arn
│     ├────────────────
│     │ each.value is object with 7 attributes
│ 
│ This object does not have an attribute named "association_policy_arn".
╵

```

Reasoning: 
Whilst most of these EC2_x and fargate roles are created automatically through Managed Node Groups, legacy, ASG or using non-module resources may result in needing to pass through roles requiring these permissions. Whislt these resources can be created external to the module, it feels more complete if this module caters for this type of access entity.

**Comment by bryantbiggs** [comment]
you don't associate policies with roles used by nodes - you use the `type` and specify `EC2_Linux` or `EC2_Windows` (the Fargate type will almost never be used by users, but its part of the API)

**Comment by Bourne-ID** [comment]
HI @bryantbiggs  - may I request to re-review this issue? Simplying the example, passing through a map of:

```hcl
access_entries = {
    role_ec2 = {
      principal_arn       = aws_iam_role.roles[1].arn
      type                = "EC2_LINUX"
    }
  }
```
into vars.access_entries results in no resources being created. 

The initial example highlighting a standard type entry + this EC2_LINUX type entry was used to demonstrate the variable was being set correctly.

**Comment by Bourne-ID** [comment]
Fixed in 20.0.1
